### ISSUE TITLE
Inexplicable post-initialization delay on NodeJS

### ISSUE THREAD

**Issue by dorkmania** [naptha/tesseract.js#450]
Tesseract v2.1.1
NodeJs v12.16.3

There seems to be a delay between 

`{ workerId: 'Worker-0-b3f2a', status: 'initialized api', progress: 1 }`

and
```
{
  workerId: 'Worker-0-b3f2a',
  jobId: 'Job-0-09a58',
  status: 'recognizing text',
  progress: 0
}
```
This delay is noticeable (~5s) even with the [example code](https://github.com/naptha/tesseract.js/blob/master/docs/examples.md#with-detailed-progress) but with a larger and/or more complex image, I can see a much larger difference (~130s).

Looking to figure out the root cause behind this and also any guidelines with respect to tuning config parameters or even input file characteristics in the interest of reducing end-to-end processing time without a significant loss in recognition accuracy. I'm sharing the [image link](https://imgur.com/a/m8hrl0o) in case it helps.

**Comment by faisal-hearth** [comment]
I'm also having the same problem, I'm trying to extract the text from a webpage screenshot (~1.6mb).

**Comment by WintrySnowman** [comment]
This may be related to [this line](https://github.com/naptha/tesseract.js/blob/adcb5b875986ebe61d7235e64d956d648c53c1dd/src/worker-script/utils/setImage.js#L12), which is causing a hang depending on how many bytes it needs to process. I've made a patch in #610 whilst adding support for `ImageData`.

**Comment by Balearica** [comment]
I noticed this as well (especially with large test images), and I believe this was resolved by #631.  Let me know if you still experience this version using the master branch.

**Comment by Balearica** [comment]
Closing as I believe this to be resolved by version 3.0.0.  Feel free to reopen or create a new issue if you encounter this problem again with the current version.
